### ISSUE TITLE
FP S1135: surrounding accented characters should be considered

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2957]
The rule currently ignores todo occurrences iff it is surrounded by a character accepted by the regular expression [A-Za-z]. This is incorrect as the rule could reports false-positives when there are surrounding accented characters, like for example:

```javascript
class C {
  // Método que calcula a média dos números
  m() { /* ... */}
}
```

Here `Método` is flagged because the _todo_ occurrence is preceded by _é_, that is, an accented character.

Reported in https://community.sonarsource.com/t/l25-typescript-catches-todo-in-part-of-other-words/56672
